### DISCUSSION TITLE
visidata won't run because of AttributeError: '_curses.curses window' object has no attribute 'get_wch'

### DISCUSSION THREAD

**Discussion by reagle** [pyenv/pyenv#2830] [Q&A]
I apologize, but I know very little about C and packages. I am not able to use run visidata with pyenv 3.12.0 (it works fine with 3.11.*). You can see the discussion thread on [visidata's github page](https://github.com/saulpw/visidata/issues/192) where @saulpw suggests that perhaps pyenv's 3.12.0 is not compiled with ncursesw? Is this true? What can I do to use visidata with 3.12.0 ?

**Comment by native-api** [comment]
> perhaps pyenv's 3.12.0 is not compiled with ncursesw

Looks like that. Note that in https://github.com/pyenv/pyenv/wiki#suggested-build-environment and https://github.com/pyenv/pyenv/wiki/Common-build-problems#prerequisites, we specifically suggest using `libncursesw` over `libncurses`.

**Reply by reagle** [reply]
Do you mean visidata should libncursesw -- or something else?

**Reply by native-api** [reply]
You need to link CPython against libncursesw when building it. Installing the suggested dependency libraries is usually enough.

**Reply by reagle** [reply]
@native-api, thank you for the response, but I don't understand. I'm a simple user wondering why visidata won't run on pyenv 3.12.0 when it will on 3.11.* -- appraently because of ncurses issues. I'm trying to figure out who should change whatever is it that they are doing that prevents this.

**Reply by native-api** [reply]
Check that you've completed the [installation instructions](https://github.com/pyenv/pyenv#installation), then [uninstall](https://github.com/pyenv/pyenv#uninstall-python-versions) and [reinstall the faulty version](https://github.com/pyenv/pyenv#install-additional-python-versions).

If the problem persists, please [report an issue](https://github.com/pyenv/pyenv/issues/new/choose) and provide the necessary diagnostic information as per the issue template. That will let us say something more specific.
